### ISSUE TITLE
Find a way to make example message skip par or message without usage of regexp 

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#13740]
Detected at https://github.com/checkstyle/checkstyle/pull/13731#discussion_r1323249392

Read it for more details

**Comment by nrmancuso** [comment]
We should try a direct substitution of `…` -> `.*` in the violation message parser. We shouldn’t have to worry about the ellipses token literal `…`, since the `.*` will just match anything anyway.

**Comment by nrmancuso** [comment]
@rnveach what do you think? I actually like the eclipses more for readability, outside of example usage. 

@romani lets move details from link to issue description and come to an agreement on what our solution will look like so we can get this approved.

**Comment by romani** [comment]
done.

**Comment by rnveach** [comment]
> may be in cases where .* is not avoidable, we will improve macros code to search replace it to ... to more natural for humans

I think if you really want to make this more natural you will have to drop the regexp completely. I have seen instances where we use something like `\'` because of the regexp.

If your replacing `.*` with `....`, then this goes against your line length argument as violations will be 2 characters longer.

It is also a danger that you will be leaving this as regexp, but replacing the regexp with still recognizable regexp characters if someone adds too many or too few periods. `.` = match any character

You may want to consider matching on `\.{4,}` for now.

I am fine with going with this change, but like I said above, regexp is what makes this not easy to read, so if you want more readable messages, you should consider replacing or hiding away the regexp completely.

**Comment by romani** [comment]
I updated description a bit to make it more clear.

Ideally we should use:
`Must include both @java.lang.Deprecated annotation and @deprecated Javadoc tag with description.`
because due to wording first and last words are very meaningful.

to keep sense but cut we can do (as we do in Inputs for test):
` // violation below 'Must include .* @deprecated Javadoc tag with description.'`

but `.*` looks weird a bit in our Example (code that we show to users) as it is unnaturaal, as regexp systax usage is not expected here at all.

so proposals is write it in more natual way:
`// violation below 'Must include ... @deprecated Javadoc tag with description.`
and internally handle a fact that ` .... ` replaced to ` .* `.

**Comment by smita1078** [comment]
I am on it.

**Comment by smita1078** [comment]
@romani , I wanted to clarify the expected action for this issue.

In the current xdocs-examples (e.g. MissingDeprecated), the violation messages already look shortened, for example:

`// violation below '@deprecated Javadoc tag with description.'
`

I could not find any occurrences of `.*` in these examples anymore.
From the issue discussion, I understand the goal is to replace regex-style `.*` with a human-friendly placeholder like `....` in user-facing examples.

However, since there are currently no` .* `occurrences in the xdoc examples, I’m unsure what the expected change is in this case.

Should we:
leave the existing manually shortened messages as they are , or
update them by reconstructing full messages and then introducing .... where parts are omitted for consistency?

Just want to confirm whether this issue is only about replacing existing `.*`, or if it also implies normalizing all xdoc violation messages to a consistent “full message + .... truncation” style.?
Thanks

**Comment by romani** [comment]
After looking on this now, I don't like .... .
It is kind of same as .* .


@smita1078 , what if we do a bit different and we explore abilities to do muliline violation message?

```
// violation 2 line below "Must include both @java.lang.Deprecated annotation
//  and @deprecated Javadoc tag with description."
```
Not closed " will means use next line as message continuation.
Or 
```
// violation 2 lines below """Must include both @java.lang.Deprecated annotation
//  and @deprecated Javadoc tag with description."""
```
Or
```
/* violation 3 lines below "Must include both @java.lang.Deprecated annotation
  and @deprecated Javadoc tag with description."
*/
```
Or 
```
/* violation 3 lines below """Must include both @java.lang.Deprecated annotation
  and @deprecated Javadoc tag with description."""
*/
```
`"""` in modern java means multiple lines string.

Or .... Feel free to suggest.
What ever is easier and reliable to implement.

**Comment by smita1078** [comment]
I agree `....` is too similar to `.*`. 
I think Option 2 (triple-quote `"""` with `//` continuation) is the cleanest because it stays consistent with the existing `//` comment style, uses delimiters, and mirrors Java text block syntax. Let me try to implement this one and see how it looks!

Also this is another way:
```
// violation 2 lines below 'Must include both @java.lang.Deprecated annotation'
//                        + 'and @deprecated Javadoc tag with description.'
```
 `+`  for continuation lines

**Comment by romani** [comment]
Just do
```
// violation 2 lines below 'Must include both @java.lang.Deprecated annotation'
//                        'and @deprecated Javadoc tag with
```

We never write comments that start from `'`.
It might conflict with multiple violations format. I think """ is easier to implement.

**Comment by smita1078** [comment]
Yes, The implementation would be in `InlineConfigParser `to detect `"""` as the message start, read continuation` //` lines stripping the `//` prefix, concatenate until closing `"""` is found, then match the assembled string against the actual violation message.

> Just do
> 
> ```
> // violation 2 lines below 'Must include both @java.lang.Deprecated annotation'
> //                        'and @deprecated Javadoc tag with
> ```
> 
> We never write comments that start from `'`. It might conflict with multiple violations format. I think """ is easier to implement.

**Comment by smita1078** [comment]
Done

> Yes, The implementation would be in `InlineConfigParser `to detect `"""` as the message start, read continuation` //` lines stripping the `//` prefix, concatenate until closing `"""` is found, then match the assembled string against the actual violation message.

**Comment by smita1078** [comment]
<img width="1423" height="627" alt="Image" src="https://github.com/user-attachments/assets/a500c6b5-7946-49d7-b8bf-83e2f58f0056" />
this looks much better. @romani  what do u think?

**Comment by romani** [comment]
@smita1078, thanks a lot for making it to happen.
